packages/infra/src/constructs/stack.ts

```diff
@@ -1,22 +1,27 @@
-import { Stack, aws_iam, aws_lambda } from 'aws-cdk-lib';
+import { Stack, type StackProps, aws_iam, aws_lambda } from 'aws-cdk-lib';
 import type { Construct } from 'constructs';
 
 import { createLambdaHookProps } from './lambda';
 
 type HookName = 'BeforeAllowTraffic' | 'AfterAllowTraffic';
 
-export type HookStackProps = {
+export type HookStackProps = StackProps & {
```

Comment:
I do find the inheritance a little 🫠 as `StackProps` could later change to conflict with the props we define on top, but that's probably unlikely for `prune` and this seems to be common-ish practice in the ecosystem 🤷